provisioning/baremetal_pod.go

```diff
@@ -277,39 +278,34 @@ func setIronicExternalIp(name string, config *metal3iov1alpha1.ProvisioningSpec)
 	}
 }
 
-func setIronicExternalUrl(config *metal3iov1alpha1.ProvisioningSpec) []corev1.EnvVar {
+func setIronicExternalUrl(client kubernetes.Interface, config *metal3iov1alpha1.ProvisioningSpec, namespace string) corev1.EnvVar {
 	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && config.VirtualMediaViaExternalNetwork {
-		envVars := []corev1.EnvVar{
-			setIronicExternalIp(externalIpEnvVar, config),
+		ipv6PodIP, err := GetPodIP(client, namespace, NetworkStackV6)
+
+		if err != nil {
```

Comment:
The IP won't be available in advance, so I assume you're relying on updating the pod after it's running?